agent/consul/state/catalog.go

```diff
@@ -752,14 +759,29 @@ func (s *Store) ServicesByNodeMeta(ws memdb.WatchSet, filters map[string]string)
 	return idx, results, nil
 }
 
+// maxIndexForService return the maximum transaction number for a service
```

Comment:
nitpick: we never refer to the Raft indexes as "transaction numbers" anywhere else so it would be less confusing to say "maximum raft index" instead.